include/hart/hart.hh

```diff
@@ -16,22 +17,27 @@ namespace sim {
 
 namespace fs = std::filesystem;
 
-using BBCache = std::unordered_map<Addr, BasicBlock>;
+class IBBCache {
+public:
+  virtual ~IBBCache() {}
+  virtual const BasicBlock &
+  lookupUpdate(Addr key, std::function<BasicBlock(Addr)> slowGetData) = 0;
+};
 
 class Hart final {
 private:
   State state_{};
   Executor exec_{};
   Decoder decoder_{};
-  BBCache cache_{};
+  std::unique_ptr<IBBCache> bbc_{};
 
   Memory &getMem() { return state_.mem; };
   Addr &getPC() { return state_.pc; };
 
   BasicBlock createBB(Addr entry);
 
 public:
-  Hart(const fs::path &executable);
+  Hart(const fs::path &executable, std::int64_t bbCacheSize);
```

Comment:
```suggestion
  Hart(const fs::path &executable, std::ssize_t bbCacheSize);
```
Why don't you use size type for size here? Just to be sure that all values will be appropriate to initialize size of the underlying container